comps/llms/rayllm/ray_serve.py

```diff
@@ -0,0 +1,514 @@
+
+        # Tweak transformer to optimize performance on Gaudi
+        adapt_transformers_to_gaudi()
+        # Not using DeepSpeed, load model locally
+        self.device = torch.device("hpu")
+        model = AutoModelForCausalLM.from_pretrained(
+            model_id_or_path, use_auth_token=use_auth_token, trust_remote_code=trust_remote_code
+        )
+        self.model = model.eval().to(self.device)
+        if self.use_hpu_graphs:
+            from habana_frameworks.torch.hpu import wrap_in_hpu_graph  # pylint: disable=E0401
+
+            self.model = wrap_in_hpu_graph(self.model)
+        else:
+            print("Warning: use_hpu_graphs is set to False. This will hurt the performance.")
+        self.tokenizer = load_tokenizer(model, model_id_or_path)
+
+    # Use dummy streamer to ignore other workers' outputs
+    def _create_dummy_streamer(self):
+        class DummyStreamer:
+            def put(self, value):
+                pass
+
+            def end(self):
+                pass
+
+        return DummyStreamer()
+
+    def _process_config(self, config):
+        config["lazy_mode"] = self.use_lazy_mode
+        config["hpu_graphs"] = self.use_hpu_graphs
+        # max_new_tokens is required for hpu
+        if "max_new_tokens" not in config:
+            config["max_new_tokens"] = 128
+
+    def get_streamer(self):
+        return TextIteratorStreamer(self.tokenizer, skip_prompt=True, timeout=0, skip_special_tokens=True)
+
+    def generate(self, prompt, **config):
+        self._process_config(config)
+
+        input_ids, input_length = self.tokenize_inputs(prompt)
+        gen_tokens = self.model.generate(input_ids, **config)
+        decode_result = self.tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)
+        if isinstance(decode_result, list) and len(decode_result) > 1:
+            return decode_result
+        elif isinstance(decode_result, list) and len(decode_result) == 1:
+            decode_result = decode_result[0]
+        return GenerateResult(
+            text=decode_result,
+            input_length=input_length,
+            generate_length=gen_tokens.size()[1] - input_length,
+        )
+
+    def streaming_generate(self, prompt, streamer, **config):
+        self._process_config(config)
+        input_ids, _ = self.tokenize_inputs(prompt)
+        self.model.generate(
+            input_ids,
+            streamer=streamer,
+            **config,
+        )
+
+
+# 1: Define a Ray Serve deployment.
+@serve.deployment
+class LLMServe:
+    _DEFAULT_MAX_BATCH_SIZE = 8
+    _DEFAULT_MAX_NUM_SEQS = 256
+
+    def __init__(
+        self, infer_conf: dict, max_batch_size=_DEFAULT_MAX_BATCH_SIZE, max_num_seqs=_DEFAULT_MAX_NUM_SEQS
+    ) -> None:
+        # All the initialization code goes here
+        self.predictor = HPUPredictor(infer_conf)
+        self.loop = asyncio.get_running_loop()
+        self.process_tool = ChatModelLLama()
```

Comment:
This won't work for all models. Need to support other models too.

Reply:
Thanks. Have updated it with more chat_processors as in the llm-on-ray.